src/VisualStudio/Core/Def/Implementation/VisualStudioSupportsFeatureService.cs

```diff
@@ -0,0 +1,106 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Editor.Shared;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Shared;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem;
+using Microsoft.VisualStudio.LanguageServices.Implementation.Venus;
+using Microsoft.VisualStudio.Text;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.SuggestionService
+{
+    internal sealed class VisualStudioSupportsFeatureService
+    {
+        [ExportWorkspaceService(typeof(ITextBufferSupportsFeatureService), ServiceLayer.Host), Shared]
+        private class VisualStudioTextBufferSupportsFeatureService : ITextBufferSupportsFeatureService
```

Comment:
these feels like could be shared as well?

Reply:
Not sure what do you mean. Each type can only `ExportWorkspaceService` once.